src/main/java/reposense/parser/CsvParser.java

```diff
@@ -150,13 +150,23 @@ private boolean isLineMalformed(CSVRecord record) {
                     csvFilePath.getFileName(), getRowContentAsRawString(record)));
             return true;
         }
-        for (String header : mandatoryHeaders()) {
-            if (get(record, header).isEmpty()) {
+
+        for (String[] equivalentHeaders : mandatoryHeaders()) {
+            boolean isLineFormatMalformed = true;
+            for (String header : equivalentHeaders) {
+                if (!get(record, header).isEmpty()) {
+                    isLineFormatMalformed = false;
+                    break;
+                }
+            }
```

Comment:
I think streams might be appropriate here. If I'm understanding this correctly, the inner for loop is equivalent to
`boolean isLineFormatMalformed = Arrays.stream(equivalentHeaders).anyMatch(...);`

Reply:
Yup, definitely better and shorter. I will use `noneMatch()` instead to remove the `!` infront as well.